### PR TITLE
feat: add logic to easily disable chains

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR adds logic that allows entire chains to be ignored via a shared ENV variable. 

Misc: I've updated the `HUB_POOL` variable on the API to remove a typo.

### CODE REVIEW

File: api/_utils.ts [hunk 1]
```diff
@@ -690,6 +707,7 @@ export function validAddress() {
 export function validAddressOrENS() {
   return define<string>("validAddressOrENS", (value) => {
     const ensDomainRegex =
+      // eslint-disable-next-line no-useless-escape
```
Comment (james-a-morris) [inline api/_utils.ts:710]: Added this to assuage the TSlint

**Comment by james-a-morris** [comment]
@mrice32 @nicholaspai Tomorrow we should be able to add `REACT_APP_DISABLED_CHAINS="10"` to Vercel and all OP routes will be disabled.

**Review by dohaki — APPROVED** [review]
LGTM 👍

File: api/_utils.ts [hunk 2]
```diff
@@ -37,8 +37,8 @@ export const gasMarkup = GAS_MARKUP ? JSON.parse(GAS_MARKUP) : {};
 // Default to no markup.
 export const DEFAULT_GAS_MARKUP = 0;
 
-// Don't permit HUP_POOL_CHAIN_ID=0
-export const HUP_POOL_CHAIN_ID = Number(REACT_APP_HUBPOOL_CHAINID || 1);
+// Don't permit HUB_POOL_CHAIN_ID=0
+export const HUB_POOL_CHAIN_ID = Number(REACT_APP_HUBPOOL_CHAINID || 1);
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply api/_utils.ts:41]: does this mean we need to change anything in Vercel before deploying this change? I recall we had an issue on prod recently where we forgot to update a Vercel variable after deploying a change to an environment variable name

File: api/_utils.ts [hunk 3]
```diff
@@ -50,11 +50,28 @@ export const DISABLED_ROUTE_TOKENS = (
...
+).split(",");
+
+const _ENABLED_ROUTES =
+  HUB_POOL_CHAIN_ID === 1
     ? enabledMainnetRoutesAsJson
     : enabledGoerliRoutesAsJson;
 
+_ENABLED_ROUTES.routes = _ENABLED_ROUTES.routes.filter(
+  ({ fromChain, toChain }) =>
+    ![fromChain, toChain].some((chainId) =>
```
Comment (nicholaspai) [inline api/_utils.ts:68]: nice style

**Review by nicholaspai — APPROVED** [review]

**Pull request merged**